packages/lib/server/getLuckyUser.ts

```diff
@@ -295,7 +295,7 @@ function filterUsersBasedOnWeights<
   // Calculate the total calibration and weight of all round-robin hosts
   let totalWeight: number;
 
-  if (attributeWeights) {
+  if (attributeWeights && attributeWeights.length > 0) {
```

Comment:
with the fix below `attributeWeights` length shouldn't be 0, but to be safe we fallback to host weights